### PR TITLE
[SDK-225] Add new classes/interfaces to support streaming export task results

### PR DESCRIPTION
**Pull request by kopreschko**
Added new ExportTask class to handle streaming of exports of arbitrary size.

### CODE REVIEW

File: labelbox/schema/data_row.py [hunk 1]
```diff
@@ -157,11 +158,26 @@ def create_attachment(self,
...
-                  task_name: Optional[str] = None,
-                  params: Optional[CatalogExportParams] = None) -> Task:
+    def export(
+        client: "Client",
+        data_rows: List[Union[str, "DataRow"]] = None,
+        global_keys: List[str] = None,
+        task_name: Optional[str] = None,
+        params: Optional[CatalogExportParams] = None,
+    ) -> ExportTask:
+        task = DataRow.export_v2(client, data_rows, global_keys, task_name,
```
[the remark this answers is not in the document]
Comment (kopreschko) [reply labelbox/schema/data_row.py:168]: - deprecation warning: yes, I think we should.
- I changed it to `Optional[List[Union[str, "DataRow"]]] = None`, also `global_keys: Optional[List[str]] = None`

File: labelbox/schema/data_row.py [hunk 2]
```diff
@@ -157,11 +158,26 @@ def create_attachment(self,
                                       res["createDataRowAttachment"])
 
     @staticmethod
-    def export_v2(client: 'Client',
-                  data_rows: List[Union[str, 'DataRow']] = None,
-                  global_keys: List[str] = None,
-                  task_name: Optional[str] = None,
-                  params: Optional[CatalogExportParams] = None) -> Task:
+    def export(
```
[the remark this answers is not in the document]
Comment (kopreschko) [reply labelbox/schema/data_row.py:161]: Descriptions added to all the new `export` methods.
I will create Documentation stories.

File: labelbox/schema/task.py [hunk 3]
```diff
@@ -62,10 +62,10 @@ def wait_till_done(self, timeout_seconds: int = 300) -> None:
         to update the task attributes.
 
         Args:
-            timeout_seconds (float): Maximum time this method can block, in seconds. Defaults to one minute.
+            timeout_seconds (float): Maximum time this method can block, in seconds. Defaults to five minutes.
         """
         check_frequency = 2  # frequency of checking, in seconds
-        while True:
+        while timeout_seconds:
```
[the remark this answers is not in the document]
Comment (kopreschko) [reply labelbox/schema/task.py:68]: Done.

File: labelbox/schema/export_task.py [hunk 4]
```diff
@@ -0,0 +1,605 @@
...
+    def type(self):
+        """Returns the type of the task."""
+        return self._task.type
+
+    def wait_till_done(self, timeout_seconds: int = 300) -> None:
+        """Waits until the task is done."""
+        return self._task.wait_till_done(timeout_seconds)
+
+    @staticmethod
+    @lru_cache(maxsize=5)
```
[the remark this answers is not in the document]
Comment (kopreschko) [reply labelbox/schema/export_task.py:520]: Definitely! That's a good catch, will look deeper into how to solve this.

File: labelbox/schema/export_task.py [hunk 5]
```diff
@@ -0,0 +1,605 @@
...
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        return False
+
+    @abstractmethod
+    def convert(self, input_args: ConverterInputArgs) -> Iterator[OutputT]:
+        """Converts the data.
+        Returns an iterator that yields the converted data.
+
+        Args:
+            current_offset: The global offset indicating the position of the data within the
```
[the remark this answers is not in the document]
Comment (kopreschko) [reply labelbox/schema/export_task.py:87]: in chars, I added this to the description.

File: labelbox/schema/export_task.py [hunk 6]
```diff
@@ -0,0 +1,605 @@
...
+    total_lines: int
+
+
+class _MetadataFileInfo(_CamelCaseMixin, BaseModel):  # pylint: disable=too-few-public-methods
+    offsets: Range
+    lines: Range
+    file: str
+
+
+@dataclass
```
[the remark this answers is not in the document]
Comment (kopreschko) [reply labelbox/schema/export_task.py:56]: I used `dataclass` for its simplicity and where more advanced features present in pydantic models weren't necessary. I also wanted to keep the code completion (IntelliSense) clean in the IDE, in the case of a dataclass, it doesn't get as cluttered compared to pydantic models.

File: labelbox/schema/export_task.py [hunk 7]
```diff
@@ -0,0 +1,605 @@
...
+                            exported files. It represents a cumulative offset across multiple
+                            files.
+            raw_data: The raw data to convert.
+
+        Yields:
+            Iterator[OutputT]: The converted data.
+        """
+
+
+class JsonConverter(Converter["JsonConverter.Output"]):  # pylint: disable=too-few-public-methods
```
[the remark this answers is not in the document]
Comment (kopreschko) [reply labelbox/schema/export_task.py]: - type check warning here:Class definition for "JsonConverter" depends on itself: I ended up separating the Output classes from the Converters.
- sdk-tests: yes, I will add those separately

File: labelbox/schema/data_row.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kopreschko) [reply labelbox/schema/data_row.py:162]: That's right, I added them.

**Review by vbrodsky — APPROVED** [review]

**Pull request merged**